scribus/scpainter.cpp

```diff
@@ -1983,9 +1999,27 @@ void ScPainter::drawText(QRectF area, QString text, bool filled, int align)
 	cairo_move_to (m_cr, x, y);
 	m_stroke.getRgbF(&r, &g, &b);
 	cairo_set_source_rgba( m_cr, r, g, b, m_stroke_trans );
+
 	for (int a = 0; a < textList.count(); ++a)
 	{
-		cairo_show_text (m_cr, textList[a].toUtf8());
+		StoryText story;
+		story.insertChars(textList[a]);
```

Comment:
I don't think that refactoring buildText will be enough because you will need styles & effects in shape() later and since we have Qstring only the TextShaper will crash soon.